src/main/java/uk/gov/ida/stub/idp/repositories/EidasSessionRepository.java

```diff
@@ -0,0 +1,11 @@
+package uk.gov.ida.stub.idp.repositories;
+
+import uk.gov.ida.common.SessionId;
+import uk.gov.ida.stub.idp.domain.EidasAuthnRequest;
+import uk.gov.ida.stub.idp.domain.IdpLanguageHint;
+
+import java.util.Optional;
+
+public interface EidasSessionRepository extends SessionRepository<EidasSession> {
+	SessionId createSession(EidasAuthnRequest eidasAuthnRequest, String relayState, Optional<IdpLanguageHint> languageHint);
```

Comment:
Given how similar EidasAuthnRequest and EidasSession are - it might make sense to make `createSession` take the generic type `T` on the abstract base class and make the caller to `createSession` construct the Session themselves. That way we can completely remove this interface.